src/pip/_internal/commands/install.py

```diff
@@ -472,16 +473,17 @@ def run(self, options: Values, args: List[str]) -> int:
             )
             env = get_environment(lib_locations)
 
+            # Display a summary of installed packages, with extra care to
+            # display a package name as it was requested by the user.
             installed.sort(key=operator.attrgetter("name"))
             items = []
-            for result in installed:
-                item = result.name
-                try:
-                    installed_dist = env.get_distribution(item)
-                    if installed_dist is not None:
-                        item = f"{item}-{installed_dist.version}"
-                except Exception:
-                    pass
+            installed_versions = {}
+            for distribution in env.iter_all_distributions():
+                installed_versions[distribution.canonical_name] = distribution.version
+            for package in installed:
+                display_name = package.name
+                version = installed_versions.get(canonicalize_name(display_name), None)
```

Comment:
While I'm not aware of a scenario where a distribution is installed and yet the metadata backend is unable to find the package, displaying `Successfully installed six-None` doesn't seem like a good idea. The old logic simply fell back to displaying only the distribution name. 

```console
(pip-20240621) ichard26@asus-ubuntu:~/dev/oss/pip$ pip install six
Collecting six
  Using cached six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: six
Successfully installed six
```

Reply:
done